be/src/vec/exec/volap_scanner.cpp

```diff
@@ -106,13 +106,11 @@ Status VOlapScanner::prepare(
                    << ", backend=" << BackendOptions::get_localhost();
                 return Status::InternalError(ss.str());
             }
-        }
-    }
 
-    {
-        // Initialize tablet_reader_params
-        RETURN_IF_ERROR(
-                _init_tablet_reader_params(key_ranges, filters, bloom_filters, function_filters));
+            // Initialize tablet_reader_params
+            RETURN_IF_ERROR(_init_tablet_reader_params(key_ranges, filters, bloom_filters,
```

Comment:
@Gabriel39 could you help to review this PR? The comment of method `_init_tablet_reader_params` says it requires caller to hold a read lock of tablet, but our code actually not.